_posts/2023-03-12-gs_00008_programming_II.md

```diff
@@ -0,0 +1,188 @@
+recommending the recipe. If the pancakes don't taste good, the instructions
+below the else statement will be executed, here, deleting the recipe.
+
+You might say that this is too easy and this can't be all. Well... Yes and No!
+Of course, there is much more to learn. Otherwise, software engineers wouldn't
+be paid that well. But with the principles presented above, you can already
+construct very complex programs.
+
+## Programming beginners FAQ
+
+Before we move on, let us answer some common questions asked by programming
+beginners.
+
+<div class='note-exercise' markdown="1">
+Which programming language should I start with?
+</div>
+
+There is no general answer to this as it highly depends on what you want to do.
+In this blog series, we will teach Python and C. Why did we choose these? Python
+is very flexible and you can do almost everything with it. You can process data,
+build graphical applications, work with artificial intelligence, and much more.
+C is, for example, an ideal choice in combination with microcontrollers that are
+often used by programming hobbyists for their home automation projects.
+Moreover, both languages are pretty well established in the community for ages.
+So, there exists an almost infinite number of public projects and you can get a
+lot of help in forums. But in general, you should decide on your project first
+and then do some research to see which language suits the best.
+
+<div class='note-exercise' markdown="1">
+Is it hard to learn a programming language?
+</div>
+
+No! You'll see that it does not take long to get your first program running.
+From there on it is just experience. The more you code the better you get. We'll
+guide you to make it as easy as possible for you.
+
+<div class='note-exercise' markdown="1">
+What is the best way to learn a programming language?
+</div>
+
+Get your fingers on a keyboard! Often we hear people say "First, I buy a book
+and learn everything." Forget about it! Programming languages have almost
+unlimited possibilities and even experienced programmers don't know all of them.
+Start with some small basic tutorials like this blog series. Then, look for your
+own project and a search engine of your choice will help you to succeed.
+
+<div class='note-exercise' markdown="1">
+Math really sucks! Can I still learn to program?
+</div>
+
+Why do people always ask this??? Of course, when you write a program that shall
+do your maths jobs, then you should know how maths works. Otherwise, it would be
+hard to write the program. Besides that, mathematical knowledge is not required.
+
+<div class='note-exercise' markdown="1">
+It sometimes takes hours to solve a problem. Is it normal?
+</div>
+
+Yes, it is! Sometimes your program simply does not do what you want and it might
+take a while to find out why. This can often be very frustrating. But this is
+not only the case for beginners. Every programmer struggles with this from time
```

Comment:
"From time to time" makes debugging sound like it does not happen very often, which we know is the opposite :D

Reply:
check https://github.com/bitsandvolts/bitsandvolts.github.io/pull/48/commits/c69f4984151fecfb6d974b40e5fee96ac47c36b9

Reply:
should be improved